hangman/interface.py

```diff
@@ -71,32 +71,30 @@ def write_hint(self, hint, theme):
 
     def write_health(self, health):
         print("\nRemaining lives: {0}".format(health))
-        print("Remaining lives: {0}".format(health))
 
     def write_guessed_letters(self, guesses):
-        print("Letters guessed: {0}".format(guesses))
+        print("Letters guessed: {0}".format(guesses)) if len(guesses) > 0 else print("Letters guessed: [ ]")
 
     def write_already_guessed(self):
         print("You have already guessed this letter! Try again!")
 
-    def write_incorrect_guess(self, guess):
-        print("\nSorry! {0} is not in the word! Try again!".format(guess))
+    def write_incorrect_guess(self, guess, health):
+        print("\nSorry! {0} is not in the word! Try again!".format(guess)) if health > 0 else print("\nSorry! {0} is not in the word!".format(guess))
```

Comment:
I don't think this change is required either because the same statement is printed for both if-else conditions?

Reply:
When health becomes 0, there are no more tries to be made, so I thought it might be nicer to not print ''Try again'' as done when there are still guessed to be made. 
![Untitled](https://user-images.githubusercontent.com/44074078/80091201-fdd26480-8560-11ea-9fe6-064f3fdc5e7f.png)

Reply:
Ah my bad, I missed the "try again" part. Instead of passing the player health into the interface (which does not seem sound from a design stand point) how about we introduce a `try` param to `def write_incorrect_guess` which can default to 1 to indicate that the player can try again? Then we move the health logic back into game.py and manipulate `try`?

Reply:
> Ah my bad, I missed the "try again" part. Instead of passing the player health into the interface (which does not seem sound from a design stand point) how about we introduce a `try` param to `def write_incorrect_guess` which can default to 1 to indicate that the player can try again? Then we move the health logic back into game.py and manipulate `try` (set it to 0 if player.health == 0)?

I added an attribute to Game() called `attempts_left` which is initialized as True and becomes Fault when player health becomes 0. So we don't pass player health logic to interface.py anymore, but a game attribute. But I'm not sure if I fully udnerstood the design issue you had with the previous (quoted) solution and if this new suggestion actually addresses it?

Reply:
It's getting there :). My suggestion was to decouple two seemingly unrelated classes from each other (player and interface). The interface class is meant to facilitate communication with the person playing the game, thus it should not rely on any info. from the `Game` and `Player` classes. So the new changes do not address the core issue (now instead of relying on Player, we rely on Game). My suggestion was to do something like this:

```python
def write_incorrect_guess(self, try=True):
    if try:
        # print something
    else:
        # print something else
```

And then from game we can do:

```python
if self.player.gethealth() > 0:
    self.interface.write_incorrect_guess()
else:
    self.interface.write_incorrect_guess(try=False)
```

This way `Interface` remains decoupled from the other classes. Hope this help! Let me know if I can help!

Reply:
Thanks for taking the time to explain, I think I understand the difference! Updated accordingly (sorry about the long commit history though)